### PR TITLE
Vote, Deposit Tokens, Withdraw Tokens page added

### PR DESCRIPTION
**Pull request by Megha-Dev-19**
- Added vote token page with deposit and withdrawal tabs
image

image

- Added vote page
image

- Added notification banner in case of success
- Added error close button

### CODE REVIEW

File: src/pages/Vote.vue [hunk 1]
```diff
@@ -40,33 +40,90 @@
...
+			const registerVoteParam: types.ExecParams = {
+				type: types.TransactionType.CallApp,
+				sign: types.SignType.SecretKey,
+				fromAccount: {
+					addr: this.walletStore.address,
+					sk: new Uint8Array(0),
+				},
+				appID: this.daoIDStore.dao_id,
+				payFlags: { totalFee: 2000 },
+				appArgs: [DAOActions.REGISTER_VOTE, `str:${this.formState.vote_type}`],
```
Comment (vuvoth) [inline src/pages/Vote.vue]: When I run this branch in my laptop, this.formState.vote_type return number type not string. It should be VoteOptions type(yes,no,abstain)

Please check it!

File: src/indexer/index.ts [hunk 2]
```diff
@@ -146,32 +146,34 @@ export const searchApplicationAndAccount = async () => {
...
 		proposalStore.setProposalAddr(lsig.address());
-
+		searchForGovASAToken();
 		if (account && account.localStateMap) {
-			daoIdStore.locked = account.localStateMap.get(
-				LOCAL_STATE_MAP_KEY.Deposit
-			) as number;
-			daoIdStore.available = account.total_amount - daoIdStore.locked;
+			daoIdStore.locked =
+				(account.localStateMap.get(LOCAL_STATE_MAP_KEY.Deposit) as number) ?? 0;
```
Comment (rmeena840) [inline src/indexer/index.ts:152]: I think this line is sufficient. `daoIdStore.locked` value will be set anyhow even `localStateMap` is present or undefined.
Just suggestion but please verify once.

File: src/indexer/index.ts [hunk 3]
```diff
@@ -181,25 +183,23 @@ export const isAssetOpted = async (address: string, asset_id: number) => {
...
-			parsedOptedApplicationInfo &&
-			parsedOptedApplicationInfo["apps-local-states"]
-		) {
-			const isGivenApplicationOpted = parsedOptedApplicationInfo[
-				"apps-local-states"
-			].find((element: any) => element["id"] === application_id);
+		if (optedApplicationInfo && optedApplicationInfo["app-local-state"]) {
+			const isGivenApplicationOpted =
+				optedApplicationInfo["app-local-state"].id === application_id;
 			if (isGivenApplicationOpted) return true;
```
Comment (vuvoth) [inline src/indexer/index.ts]: We only need to return `optedApplicationInfo["app-local-state"].id === application_id;`

File: src/indexer/index.ts [hunk 4]
```diff
@@ -146,32 +146,34 @@ export const searchApplicationAndAccount = async () => {
...
-			daoIdStore.locked = account.localStateMap.get(
-				LOCAL_STATE_MAP_KEY.Deposit
-			) as number;
-			daoIdStore.available = account.total_amount - daoIdStore.locked;
+			daoIdStore.locked =
+				(account.localStateMap.get(LOCAL_STATE_MAP_KEY.Deposit) as number) ?? 0;
+		} else {
+			daoIdStore.locked = 0;
 		}
+		daoIdStore.available = account.total_amount - daoIdStore.locked;
```
Comment (rmeena840) [inline src/indexer/index.ts]: Can total_amount be undefined? It might give runtime error if we get undefined error.

File: src/pages/Proposal.vue [hunk 5]
```diff
@@ -268,27 +270,17 @@ export default defineComponent({
 					},
 				];
 				let response = await this.walletStore.webMode.executeTx(addProposalTx);
+				openSuccessNotificationWithIcon(
+					"Success",
+					"Your Proposal has benn created."
```
Comment (rmeena840) [inline src/pages/Proposal.vue]: Please move msg to new file inside success.constant.ts

File: src/pages/Proposal.vue [hunk 6]
```diff
@@ -181,16 +189,10 @@ export default defineComponent({
 				} = values;
 				if (typeof this.daoStore.dao_id === "undefined") {
```
Comment (vuvoth) [inline src/pages/Proposal.vue]: We duplicated code when check error in Proposal and Vote
https://github.com/scale-it/sigmadao-ui/blob/ac184aaaeb2fb55e7b8489413405fcc7104eb629/src/pages/DepositToken.vue#L77

File: src/pages/Vote.vue [hunk 7]
```diff
@@ -39,34 +46,86 @@
...
 		return {
 			formState,
+			daoIDStore,
+			walletStore,
 			validateMessages: VALIDATE_MESSAGES,
 		};
 	},
+	methods: {
+		async onFinish() {
+			if (typeof this.daoIDStore.dao_id === "undefined") {
```
Comment (vuvoth) [inline src/pages/Vote.vue]: Also duplicated code at here.
Could we remove duplicated code?

**Review by vuvoth — COMMENTED** [review]
Nice work! I left few comments.

**Review by rmeena840 — CHANGES_REQUESTED** [review]

File: src/indexer/index.ts (hunk 3 above)
Comment (Megha-Dev-19) [reply src/indexer/index.ts]: oh, yes that will do it, will update

File: src/pages/Proposal.vue (hunk 6 above)
Comment (Megha-Dev-19) [reply src/pages/Proposal.vue]: yes, error conditions are repeated at multiple places, will create a common function and return error from there

File: src/indexer/index.ts (hunk 2 above)
Comment (Megha-Dev-19) [reply src/indexer/index.ts:152]: But we are setting the `locked` value from `localStateMap` as it has a deposit key, so how will it be updated if it is undefined..?  
May be I m not able to understand what you mean. Can you please elaborate.?

File: src/indexer/index.ts (hunk 4 above)
Comment (Megha-Dev-19) [reply src/indexer/index.ts]: Actually I have to remove this, `total_amount` is algos but we needed to show gov-token so I updated the UI but forgot to remove this line, will update

File: src/pages/Proposal.vue (hunk 5 above)
Comment (Megha-Dev-19) [reply src/pages/Proposal.vue]: okay sure

**Review by vuvoth — APPROVED** [review]
LGTM!

File: src/constants/error.constant.ts [hunk 8]
```diff
@@ -1,2 +1,40 @@
+import DaoID from "@/store/DaoID";
+import WalletStore from "@/store/WalletStore";
+import { message, notification } from "ant-design-vue";
+import { reactive } from "vue";
+
 export const APP_NOT_FOUND = "Please add DAO App ID";
 export const TOKEN_NOT_FOUND = "Govt token not found";
+export const WALLET_NOT_CONNECT = "Please connect to your Wallet";
+
+export const OverallErrorCheck = () => {
```
Comment (rmeena840) [inline src/constants/error.constant.ts]: naming convention of method name.

**Review by rmeena840 — APPROVED** [review]
Left one comment. Pre-approving.

File: src/constants/error.constant.ts (hunk 8 above)
Comment (Megha-Dev-19) [reply src/constants/error.constant.ts]: updated

**Pull request merged**